ui/lesson/OpRunner/OpRunner.tsx

```diff
@@ -1,9 +1,48 @@
-import React, { useEffect, useState, useRef, useCallback } from 'react'
+import React, { useEffect, useState, useRef } from 'react'
 import clsx from 'clsx'
 import LanguageExecutor from './LanguageExecutor'
 import _ from 'lodash'
 import { StatusBar } from 'ui/common'
 import { MainState, OpRunnerTypes, StackType, T } from './runnerTypes'
+import { ArcherElement } from 'react-archer'
+import { RelationType } from 'react-archer/lib/types'
+import { useArrows } from 'state/ArrowsContext'
+import useHorizontalScroll from 'hooks/useHorizontalScroll'
+
+const arrowLineStyles = {
+  startMarker: true,
+  endMarker: false,
+  strokeColor: '#3DCFEF',
+}
+
+const getRelationsForOperation = (
+  operation: string,
+  currentIndex: number
+): RelationType[] => {
+  switch (operation) {
+    case 'OP_ADD':
+    case 'OP_SUB':
+    case 'OP_MUL':
+    case 'OP_DIV':
+    case 'OP_DUP':
```

Comment:
These are all the op codes I could think of which could use arrows right now - are there any others?

Reply:
Yh some like OP_EQUAL, OP_PUSH

Reply:
Yeah I think realistically we will probably want arrows an all the opcodes other than the constants OP_1 OP_TRUE etc... I'm not sure we actually want OP_PUSH since that is taking stuff from the script and not the stack so it behaves similarly to a constant.